app/resources/v1/users.js

```diff
@@ -195,90 +230,288 @@ exports.post = function (req, res) {
+      if (data) {
+        user.profileImageUrl = data.twitter_profile_image_url
+      } else {
+        user.profileImageUrl = user.profile_image_url
+      }
+      res.status(200).send(asUserJson(data || user))
+    }
+
+    let responseAlreadySent = false
+
+    const handleFetchUserProfileFromTwitter = function (err, res) {
+      if (err) {
+        logger.error('Twitter API call users/show returned error.')
+        logger.error(err)
+      }
+
+      if (responseAlreadySent) {
+        logger.debug(
+          { profile_image_url: res.profile_image_url },
+          'Twitter API users/show call returned but response already sent!'
+        )
+      } else {
+        logger.debug(
+          { profile_image_url: res.profile_image_url },
+          'Twitter API users/show call returned. Sending response with Twitter data.'
+        )
+        responseAlreadySent = true
+
+        if (!res) {
+          logger.error('Twitter API call users/show did not return any data.')
+        }
+
+        sendUserJson({
+          twitter_profile_image_url: res.picture
+        })
+      }
+    } // END function - handleFetchUserProfileFromTwitter
+
+    if (twitterApiClient && !user.profile_image_url) {
+      logger.debug(
+        'About to call Twitter API: /users/show.json?user_id=' + user.twitter_id
+      )
+      twitterApiClient.get(
+        '/users/show.json',
+        { user_id: user.twitter_id },
+        handleFetchUserProfileFromTwitter
+      )
+      setTimeout(function () {
+        if (!responseAlreadySent) {
+          logger.debug(
+            'Timing out Twitter API call after %d milliseconds and sending partial response.',
+            config.twitter.timeout_ms
+          )
+          responseAlreadySent = true
+          sendUserJson()
+        }
+      }, config.twitter.timeout_ms)
+    } else {
+      sendUserJson()
+    }
+  } // END function - handleFindUser
+
+  const handleError = function (error) {
+    console.log('handleError???', error)
+    switch (error) {
+      case ERRORS.USER_NOT_FOUND:
+        res.status(404).json({ status: 404, msg: 'User not found.' })
+        return
+      case ERRORS.CANNOT_GET_USER:
+        res.status(500).json({ status: 500, msg: 'Error finding user.' })
+        return
+      case ERRORS.UNAUTHORISED_ACCESS:
+        res
+          .status(401)
+          .json({ status: 401, msg: 'User with that login token not found.' })
+        return
+      default:
+        res.status(500).end()
+    }
   }
 
   const findUserById = async function (userId) {
     let user
-
     try {
-      user = await User.findOne({ id: userId })
+      user = await User.findOne({ where: { id: userId } })
     } catch (err) {
       logger.error(err)
       throw new Error(ERRORS.CANNOT_GET_USER)
     }
 
+    // if enabled, returns 401 for any attempt to get another user's data
+    // if (user.login_tokens.indexOf(req.loginToken) === -1) {
+    //   res.status(401).end()
+    //   return
+    // }
```

Comment:
Why would returning 401 be disabled?

Reply:
This feature gets used when we look at another user's street gallery, for example. Do we want to disable that? I think this was originally written in `users_pg` under that impression but it doesn't seem to be how we're working now.